src/pyhf/pdf.py

```diff
@@ -538,6 +538,30 @@ def logpdf(self, pars, data):
         try:
             tensorlib, _ = get_backend()
             pars, data = tensorlib.astensor(pars), tensorlib.astensor(data)
+            # Verify parameter and data shapes
+            try:
+                if pars.shape[-1] != len(self.config.suggested_init()):
+                    raise ValueError
```

Comment:
Using `ValueError` as a standin for now. Can implement a better pyhf exception later in the PR.

Reply:
I would not catch the exception just rause 

```
raise ValueError('eval failed as pars has len {} but {} was expected'.format(...))

...

raise ValueError('eval failed as data has len {} but {} was expected'.format(...))
```